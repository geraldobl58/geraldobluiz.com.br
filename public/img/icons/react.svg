<svg width="85" height="85" viewBox="0 0 85 85" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M42.5 60.2084C19.6562 60.2084 1.77081 52.4167 1.77081 42.5C1.77081 32.5834 19.6562 24.7917 42.5 24.7917C65.3437 24.7917 83.2291 32.5834 83.2291 42.5C83.2291 52.4167 65.3437 60.2084 42.5 60.2084ZM42.5 28.3334C20.1875 28.3334 5.31248 35.5938 5.31248 42.5C5.31248 49.4063 20.1875 56.6667 42.5 56.6667C64.8125 56.6667 79.6875 49.4063 79.6875 42.5C79.6875 35.5938 64.8125 28.3334 42.5 28.3334Z" fill="#5C5C5E"/>
<path d="M26.7396 78.9792C24.9688 78.9792 23.5521 78.625 22.1354 77.7396C13.4584 72.7813 15.7604 53.4792 27.0938 33.6459C32.4063 24.4375 38.9584 16.6459 45.3334 11.6875C52.2396 6.37503 58.4375 4.78128 62.6875 7.26045C67.1146 9.73962 68.7084 15.9375 67.6459 24.6146C66.5834 32.7604 63.0417 42.3229 57.7292 51.5313C52.4167 60.7396 45.8646 68.5313 39.4896 73.4896C34.8854 77.0313 30.4584 78.9792 26.7396 78.9792ZM58.2604 9.56253C55.4271 9.56253 51.7084 11.1563 47.6354 14.3438C41.6146 19.125 35.4167 26.5625 30.2813 35.4167C19.125 54.7188 18.0625 71.3646 23.9063 74.7292C26.9167 76.5 31.875 74.9063 37.3646 70.6563C43.3854 65.875 49.5834 58.4375 54.7188 49.5834C59.8542 40.7292 63.2188 31.698 64.2813 24.0834C65.1667 17 64.1042 12.0417 61.0938 10.2709C60.2084 9.9167 59.3229 9.56253 58.2604 9.56253Z" fill="#5C5C5E"/>
<path d="M58.4375 78.9792C49.5833 78.9792 36.8333 68.1771 27.2708 51.3542C15.7604 31.5208 13.4583 12.2188 22.1354 7.26042C30.8125 2.30209 46.3958 13.8125 57.9062 33.6458C63.2187 42.8542 66.7604 52.4167 67.8229 60.5625C69.0625 69.2396 67.2916 75.2604 62.8646 77.9167C61.4479 78.625 60.0312 78.9792 58.4375 78.9792ZM23.9062 10.2708C18.0625 13.6354 19.125 30.2813 30.2812 49.5833C41.4375 68.8854 55.25 78.0938 61.0937 74.7292C64.1041 72.9583 65.1666 68 64.2812 60.9167C63.2187 53.3021 59.8541 44.2708 54.7187 35.4167C43.5625 16.1146 29.75 6.90625 23.9062 10.2708Z" fill="#5C5C5E"/>
<path d="M42.5 49.5834C46.412 49.5834 49.5834 46.412 49.5834 42.5C49.5834 38.588 46.412 35.4167 42.5 35.4167C38.588 35.4167 35.4167 38.588 35.4167 42.5C35.4167 46.412 38.588 49.5834 42.5 49.5834Z" fill="#5C5C5E"/>
</svg>
