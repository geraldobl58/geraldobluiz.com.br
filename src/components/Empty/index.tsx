import Link from 'next/link'

import * as S from './styles'

export type EmptyProps = {
  title: string
  description: string
  hasLink?: boolean
}

const Empty = ({ title, description, hasLink }: EmptyProps) => (
  <S.Wrapper>
    <S.Image
      src="/img/empty.svg"
      alt="Whoops: Página não encontrada!"
      role="image"
    />

    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    {hasLink && (
      <Link href="/" passHref>
        <S.LinkHref>Ir para a home</S.LinkHref>
      </Link>
    )}
  </S.Wrapper>
)

export default Empty
