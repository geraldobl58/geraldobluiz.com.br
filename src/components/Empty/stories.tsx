import { Story, Meta } from '@storybook/react/types-6-0'
import Empty, { EmptyProps } from '.'

export default {
  title: 'Empty',
  component: Empty
} as Meta

export const Default: Story<EmptyProps> = (args) => <Empty {...args} />

Default.args = {
  title: 'Whoops: página não encontrada!',
  description: 'Volte e confira mais sobre meu portfolio',
  hasLink: true
}
