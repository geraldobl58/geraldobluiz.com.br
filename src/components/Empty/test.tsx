import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Empty from '.'

const props = {
  title: 'titulo default',
  description: 'descricao default'
}

describe('<Empty />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Empty {...props} hasLink />)

    expect(
      screen.getByRole('image', { name: /Whoops: Página não encontrada!/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /titulo default/i })
    ).toBeInTheDocument()
    expect(screen.getByText(/descricao default/i)).toBeInTheDocument()

    expect(
      screen.getByRole('link', { name: /ir para a home/i })
    ).toHaveAttribute('href', '/')

    expect(container.parentElement).toMatchSnapshot()
  })

  it('should not render link when haslink is not passed', () => {
    renderWithTheme(<Empty {...props} />)

    expect(
      screen.queryByRole('link', { name: /ir para a home/i })
    ).not.toBeInTheDocument()
  })
})
