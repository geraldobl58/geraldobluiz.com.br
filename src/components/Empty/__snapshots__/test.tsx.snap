// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c1 {
  max-width: 100%;
  width: 30rem;
}

.c2 {
  color: #FF9901;
  font-size: 5.6rem;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 400;
  margin-bottom: 3.2rem;
}

.c4 {
  color: #FAFAFA;
}

<body>
  <div>
    <main
      class="c0"
    >
      <img
        alt="Whoops: Página não encontrada!"
        class="c1"
        role="image"
        src="/img/empty.svg"
      />
      <h2
        class="c2"
      >
        titulo default
      </h2>
      <p
        class="c3"
      >
        descricao default
      </p>
      <a
        class="c4"
        href="/"
      >
        Ir para a home
      </a>
    </main>
  </div>
</body>
`;
